File: src/Tiantong.Wms.Api/Entities/PurchaseOrderItem.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tiantong.Wms.Api
{
  [Table("purchase_order_items")]
  public class PurchaseOrderItem : Entity
  {
    [Key]
    public int id { get; set; }

    public int good_id { get; set; }

    public int item_id { get; set; }

    public double price { get; set; }

    public int quantity { get; set; }

    public string delivery_cycle { get; set; }

    public string tax_number { get; set; }

    public string tax_name { get; set; }

    public string tax_specification { get; set; }

    public string tax_type { get; set; }

    public double tax_rate { get; set; }

    public int[] item_project_ids { get; set; }
  }
}

File: src/Tiantong.Wms.Client/src/Entities/PurchaseOrderItem.ts
import PurchaseOrderItemProject from './PurchaseOrderItemProject'
import Invoice from './Invoice'

export default class PurchaseOrderItem {
  id: number = 0

  order_id: number = 0

  good_id: number = 0

  item_id: number = 0

  index: number = 0

  price: number = 0

  quantity: number = 0

  comment: string = ''

  delivery_cycle: string = '30天'

  arrived_at: string = ''

  invoice: Invoice

  projects: Array<PurchaseOrderItemProject>

  constructor (projects: Array<PurchaseOrderItemProject> = []) {
    this.projects = projects
    this.invoice = new Invoice
  }
}
